File: valid-palindrome/valid_palindrome_test.go
package validpalindrome

import (
	"testing"
)

func TestValidPalindrome(t *testing.T) {
	tests := []struct {
		name string
		in   string
		want bool
	}{
		{
			name: "lowercase palindrome",
			in:   "level",
			want: true,
		},
		{
			name: "not palindrome",
			in:   "algorithm",
			want: false,
		},
		{
			name: "palindrome with non-letter chars",
			in:   "A man, a plan, a canal: Panama.",
			want: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			if got := Palindrome(tt.in); got != tt.want {
				t.Errorf("Palindrome(%s) = %v, want %v", tt.in, got, tt.want)
			}
		})
	}
}

File: week-01/valid-palindrome/valid_palindrome_test.py
from valid_palindrome import valid_palindrome


def test_valid_palindrome():
    tests = [
        {
            "input": "",
            "want": True,
        },
        {
            "input": "tattarrattat",
            "want": True,
        },
        {
            "input": "coding",
            "want": False,
        },
    ]

    for test in tests:
        assert valid_palindrome(test["input"]) == test["want"]
